contexts/FinishedBooksContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { createContext, useState } from "react";
+
+export const FinishedBooks = createContext();
+
+function FinishedBooksProvider(props) {
+  const [finishedBooks, setFinishedBooks] = useState([]);
+  const fetchFinishedBooks = () => {
+    fetch("http://localhost:3000/api/v1/finished_books", {
+      method: "get",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: localStorage.getItem("token"),
+      },
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        setFinishedBooks(data);
+        return data;
+      });
+  };
+
+  return (
+    <FinishedBooks.Provider
+      value={{ finishedBooks, setFinishedBooks, fetchFinishedBooks }}
+    >
+      {props.children}
+    </FinishedBooks.Provider>
+  );
+}
+
+export default FinishedBooksProvider;

```

Comment:
Please reduce amount of times that you need to do `export default`. For next.js pages it's fine (since it is by design), but for regular components, please do a simple `export`. It's easier to autocomplete if it's using regular export

```suggestion
export { FinishedBooksProvider };
```